src/compiler/parser.rs

```diff
@@ -1,16 +1,40 @@
 #![allow(non_upper_case_globals)]
 
-use std::sync::{Mutex, MutexGuard};
+use std::rc::Rc;
 
 use crate::{
     create_detached_diagnostic, create_node_factory, create_scanner,
     get_binary_operator_precedence, is_same_variant, last_or_undefined, normalize_path,
-    object_allocator, BaseNode, BaseNodeFactory, BinaryExpression, Debug_, DiagnosticMessage,
-    DiagnosticWithDetachedLocation, Diagnostics, Expression, Identifier, LiteralLikeNode, Node,
-    NodeArray, NodeArrayOrVec, NodeFactory, NodeInterface, OperatorPrecedence, Scanner, SourceFile,
+    object_allocator, set_text_range_pos_end, BaseNode, BaseNodeFactory, BinaryExpression, Debug_,
+    DiagnosticMessage, DiagnosticRelatedInformationInterface, DiagnosticWithDetachedLocation,
+    Diagnostics, Expression, Identifier, LiteralLikeNode, Node, NodeArray, NodeArrayOrVec,
+    NodeFactory, NodeInterface, OperatorPrecedence, ReadonlyTextRange, Scanner, SourceFile,
     Statement, SyntaxKind,
 };
 
+fn visit_node<TNodeCallback: FnMut(Rc<Node>)>(mut cb_node: TNodeCallback, node: Rc<Node>) {
+    cb_node(node)
+}
+
+pub fn for_each_child<TNodeCallback: FnMut(Rc<Node>), TNodesCallback: FnMut(&NodeArray)>(
```

Comment:
This appears to be sort of a generic AST node visitor helper

Reply:
Plain `Fn` seems to work rather than `FnMut` for both of these.